### ISSUE TITLE
No test logs if TestNG Test has timeout parameter

### ISSUE THREAD

**Issue by violant** [reportportal/client-java#105]
I faced with no logs in RP when I specify timeout parameter in TestNG Test annotation. All events that are relayed to start/stop/methods reflected fine in RP, but no logging from TestNG test that I send through RP appender. 
Here is example of `@Test` annotation
`@Test(description = "test", dataProvider = "DataProvider", alwaysRun = true, timeout = 60)`

in DEBUG MODE
`event.getmessage()` has correct message, but
in `com.epam.reportportal.service.ReportPortal.emitLog` next variable is null
```
LoggingContext loggingContext = LoggingContext.CONTEXT_THREAD_LOCAL.get();
loggingContext = null
```

after removing timeout parameter all work well

**Comment by HardNorth** [comment]
As a possible solution we can use InheritedThreadLocal for every Thread, except main. This will probably work for TestNG and JUnit5, since the first one is using main thread only for Suite creation and the second one is not using main thread at all. But what if some frameworks use main thread in the same way as other threads?

**Comment by dagansandler** [comment]
@HardNorth is there any progress on this? Perhaps this can be parameterized in the client and defaulted to the existing behavior? and frameworks that need this behavior can update to use this setting to solve this issue.

**Comment by HardNorth** [comment]
@dagansandler not yet, the issue has milestone 6 since it requires serious logging logic rewrite. Currently I'm working on milestone 5.1, which is related to Android compatibility.

**Comment by HardNorth** [comment]
Fixed in #183
